### PR TITLE
Update hpxcxx to look at lib64 and use python3

### PR DESCRIPTION
**Pull request by stevenrbrandt**
Fixes issues using hpxcxx on the singularity image

## Proposed Changes

  - check explicitly for the hpx_application.pc file
  - use python3
  - check lib64 in addition to lib

### CODE REVIEW

File: cmake/HPX_GeneratePackage.cmake [hunk 1]
```diff
@@ -53,6 +53,12 @@ foreach(dir ${HPX_LIBRARY_DIR})
   set(HPX_PKG_LIBRARY_DIR "${HPX_PKG_LIBRARY_DIR} -L${dir}")
 endforeach()
 
+find_program(FULL_PYTHON_CMD python3)
```
Comment (hkaiser) [inline cmake/HPX_GeneratePackage.cmake]: We already have `PYTHON_EXECUTABLE` set to the python interpreter found by the build system.

Comment (hkaiser) [reply cmake/HPX_GeneratePackage.cmake]: Also, we shouldn't explicitly limit things to Python V3.

**Comment by stevenrbrandt** [comment]
@hkaiser it never limited things to Python v3, but it no longer matters because it now uses @PYTHON_EXECUTABLE@.

**Comment by msimberg** [comment]
Is it a must to use the exact path of the Python executable? This means that the path will be system specific, which can be annoying for cross-compilation. Not a huge deal, but if it's not necessary I'd rather not hardcode it like that.

**Comment by hkaiser** [comment]
#3718 was asking for exactly this. I'm not sure what's the best way out...

**Pull request merged**